debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java

```diff
@@ -2381,6 +2382,44 @@ public void shouldProduceMessagesOnlyForConfiguredTables() throws Exception {
         stopConnector();
     }
 
+    @Test
+    @FixFor("DBZ-2699")
+    public void shouldEmitNoEventsForSkippedUpdateAndDeleteOperations() throws Exception {
+        TestHelper.dropDefaultReplicationSlot();
+        TestHelper.execute(SETUP_TABLES_STMT);
+        Configuration config = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.NEVER.getValue())
+                .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                .with(PostgresConnectorConfig.SKIPPED_OPERATIONS, "u,d")
+                .build();
+        start(PostgresConnector.class, config);
+        assertConnectorIsRunning();
+        waitForStreamingRunning("postgres", TestHelper.TEST_SERVER);
+        assertNoRecordsToConsume();
+
+        TestHelper.execute("INSERT INTO s2.a VALUES(201, 201, 'insert-201')");
+        TestHelper.execute("UPDATE s2.a SET bb='update-201' WHERE pk=201");
+        TestHelper.execute("INSERT INTO s2.a VALUES(202, 202, 'insert-202')");
+        TestHelper.execute("DELETE FROM s2.a WHERE pk=202");
+        TestHelper.execute("INSERT INTO s2.a VALUES(203, 203, 'insert-203')");
+
+        SourceRecords records = consumeRecordsByTopic(3);
+        List<SourceRecord> recordsForTopic = records.recordsForTopic(topicName("s2.a"));
+
+        assertThat(recordsForTopic.size()).isEqualTo(3);
+        assertInsert(recordsForTopic.get(0), PK_FIELD, 201);
+        assertInsert(recordsForTopic.get(1), PK_FIELD, 202);
```

Comment:
hey @blcksrx. I think the current failing test comes from the test that you added.

`recordsForTopic.get(1)` comes from `TestHelper.execute("UPDATE s2.a SET bb='update-201' WHERE pk=201");` statement which is an update and not a CREATE/INSERT.